stdlib/test/os/path/test_expandvars.mojo

```diff
@@ -0,0 +1,80 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+import os
+from os.path import expandvars
+from testing import assert_equal
+
+
+def test_expansion():
+    _ = os.setenv("TEST_VAR", "World")
```

Comment:
**Suggestion** It may be nice to unset these variables explicitly at the end of each function to avoid state carrying over to the next set of tests within this file.  I think it would fall under the "element of least surprise" sort of thing - WDYT?  You could also introduce a local context manager whos `__exit__` function unsets the env variable if that's any cleaner.

Reply:
Done 👍🏾

We didn't have `os.unsetenv` yet, so I added that function with a test. If it's preferable to wrap up the `external_call` part of that function and move that over into `sys._libc` I can do that as well.